src/main/java/com/epam/grid/engine/provider/utils/slurm/job/SacctCommandParser.java

```diff
@@ -0,0 +1,127 @@
+/*
+ *
+ *  * Copyright 2022 EPAM Systems, Inc. (https://www.epam.com/)
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ *
+ */
+
+package com.epam.grid.engine.provider.utils.slurm.job;
+
+import com.epam.grid.engine.entity.job.slurm.SlurmJob;
+import com.epam.grid.engine.exception.GridEngineException;
+import com.epam.grid.engine.provider.utils.slurm.DateUtils;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.springframework.http.HttpStatus;
+
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.epam.grid.engine.utils.TextConstants.COMMA;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class SacctCommandParser {
+
+    private static final String STANDARD_SLURM_DELIMETER = "\\|";
+    private static final String FAILED_TO_PARSE_JOB_DATA = "failed to parse job data";
+    private static final String SLURM_DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss";
+
+    public static List<String> parseSlurmJob(final String jobString,
+                                             final int fieldsCount) {
+        final String[] jobArray = jobString.split(STANDARD_SLURM_DELIMETER);
+        if (jobArray.length != fieldsCount) {
+            throw new GridEngineException(HttpStatus.BAD_REQUEST, "An error occurred during job data parsing");
+
+        }
+        return Arrays.stream(jobArray).collect(Collectors.toList());
+    }
+
+    public static SlurmJob mapJobDataToSlurmJob(final List<String> jobData) {
+        try {
+            return SlurmJob.builder()
+                    .account(jobData.get(0))
+                    .tresPerNode(jobData.get(1))
+                    .minCpus(Integer.parseInt(jobData.get(2)))
+                    .minTmpDisk(Integer.parseInt(jobData.get(3)))
+                    .endTime(!jobData.get(4).equals("N/A")
```

Comment:
Parsing procedure for that field, `startTime` and `submissionTime` seems to be the same - can it be generalized and moved to a separate `SacctCommandParser.convertStringToTimeEntry` method?

Reply:
fixed